solidity/test/BondedECDSAKeepIntegrationTest.js

```diff
@@ -72,20 +74,93 @@ contract("BondedECDSAKeepFactory", async (accounts) => {
         threshold,
         keepOwner,
         bond,
+        stakeLockDuration,
         {from: application, value: feeEstimate}
       )
 
-      await keepFactory.openKeep(groupSize, threshold, keepOwner, bond, {
-        from: application,
-        value: feeEstimate,
-      })
+      await keepFactory.openKeep(
+        groupSize,
+        threshold,
+        keepOwner,
+        bond,
+        stakeLockDuration,
+        {
+          from: application,
+          value: feeEstimate,
+        }
+      )
 
       assert.equal(
         await tokenStaking.getAuthoritySource(keepAddress),
         keepFactory.address,
         "invalid token staking authority source"
       )
     })
+
+    it("locks members token stakes", async () => {
+      const tx = await keepFactory.openKeep(
+        groupSize,
+        threshold,
+        keepOwner,
+        bond,
+        stakeLockDuration,
+        {
+          from: application,
+          value: feeEstimate,
+        }
+      )
+      const keepAddress = tx.logs[0].args.keepAddress
+
+      const expectedExpirationTime = (await time.latest()).add(
+        stakeLockDuration
+      )
+
+      for (let i = 0; i < members.length; i++) {
+        const {creators, expirations} = await tokenStaking.getLocks(members[i])
+
+        assert.deepEqual(
+          creators,
+          [keepAddress],
```

Comment:
`deepEqual` will cover both the length and values assertion, so why spliting it into two?
For exceptions we do `expirations[0]` because we need to compare BNs.

Reply:
Ah, this is `deepEqual`. Alright, let's leave it as is.